### PR TITLE
Pubsub Changes for JS

### PR DESCRIPTION
**Pull request by AliabbasMerchant**

### CODE REVIEW

**Comment by YourTechBud** [comment]
You should take the subject and the payload in the body. Lets avoid the `.` all together.

File: model/pubsub.go [hunk 1]
```diff
@@ -24,3 +24,8 @@ type PubsubMsgResponse struct {
 	Error  string `json:"error"`
 	Msg    []byte `json:"msg"`
 }
+
+type PubsubPublishRequest struct {
+	Subject string `json:"subject"`
+	Data    string `json:"data"`
```
Comment (YourTechBud) [inline model/pubsub.go]: Shouldn't this be an interface?

File: utils/handlers/pubsub.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ func HandlePublishCall(pubsub *pubsub.Module, auth *auth.Module) http.HandlerFun
 		}
 		token := strings.TrimPrefix(tokens[0], "Bearer ")
 
-		status, err := pubsub.Publish(project, token, subject, b)
+		status, err := pubsub.Publish(project, token, req.Subject, []byte(req.Data))
```
Comment (YourTechBud) [inline utils/handlers/pubsub.go]: Dont take a byte array as a parameter, take an interface. You should marshal inside.

**Comment by AliabbasMerchant** [comment]
@YourTechBud Done

File: utils/handlers/pubsub.go (hunk 2 above)
Comment (YourTechBud) [reply utils/handlers/pubsub.go]: Parse an interface directly here. No need to marshal again inside the Publish() function

**Comment by AliabbasMerchant** [comment]
@YourTechBud Done

**Pull request merged**